packages/core/admin/admin/src/components/PageHelpers.tsx

```diff
@@ -0,0 +1,195 @@
+
+  return (
+    <Main height="100%">
+      <Flex alignItems="center" height="100%" justifyContent="center">
+        <EmptyStateLayout
+          icon={<EmptyPermissions width="10rem" />}
+          content={formatMessage({
+            id: 'app.components.EmptyStateLayout.content-permissions',
+            defaultMessage: "You don't have the permissions to access that content",
+          })}
+          {...props}
+        />
+      </Flex>
+    </Main>
+  );
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * Protect
+ * -----------------------------------------------------------------------------------------------*/
+export interface ProtectProps {
+  /**
+   * The children to render if the user has the required permissions.
+   * If providing a function, it will be called with an object containing
+   * the permissions the user has based on the array you passed to the component.
+   */
+  children: React.ReactNode | ((args: { permissions: Permission[] }) => React.ReactNode);
+  /**
+   * The permissions the user needs to have to access the content.
+   */
+  permissions?: Array<Omit<Partial<Permission>, 'action'> & Pick<Permission, 'action'>>;
+}
+
+/**
+ * @public
+ * @description A wrapper component that should be used to protect a page. It will check the permissions
+ * you pass to it and render the children if the user has the required permissions. If a user does not have ALL
+ * the required permissions, it will redirect the user to the home page. Whilst these checks happen it will render
+ * the loading component and should the check fail it will render the error component with a notification.
+ */
+const Protect = ({ permissions = [], children }: ProtectProps) => {
+  const { allPermissions } = useRBACProvider();
+  const toggleNotification = useNotification();
+  const { _unstableFormatAPIError: formatAPIError } = useAPIErrorHandler();
+
+  const matchingPermissions = allPermissions.filter(
+    (permission) =>
+      permissions.findIndex(
+        (perm) => perm.action === permission.action && perm.subject === permission.subject
+      ) >= 0
+  );
+
+  const shouldCheckConditions = matchingPermissions.some(
+    (perm) => Array.isArray(perm.conditions) && perm.conditions.length > 0
+  );
+
+  const {
+    isLoading,
+    error,
+    data = [],
+  } = useCheckPermissionsQuery(
+    {
+      permissions: matchingPermissions.map((perm) => ({
+        action: perm.action,
+        subject: perm.subject,
+      })),
+    },
+    {
+      skip: !shouldCheckConditions,
+    }
+  );
+
+  React.useEffect(() => {
+    if (error) {
+      toggleNotification({
+        type: 'warning',
+        message: formatAPIError(error),
+      });
+    }
+  }, [error, formatAPIError, toggleNotification]);
+
+  if (isLoading) {
+    return <Loading />;
+  }
+
+  if (error) {
+    return <Error />;
+  }
+
+  const canAccess = shouldCheckConditions
+    ? data.every((datum) => datum === true)
```

Comment:
`? !data.includes(false)`

Reply:
I tried to play with permissions, but I got `warning: policy failed` everytime, could not see no permission message

Reply:
`!data.includes(false)` yeah thats nicer!

Reply:
> I tried to play with permissions, but I got warning: policy failed everytime, could not see no permission message

This is because we were still calling the API even if you don't have the read permission, should be fixed now!